WordPress/src/main/java/org/wordpress/android/util/audio/AudioRecorder.kt

```diff
@@ -0,0 +1,168 @@
+package org.wordpress.android.util.audio
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager
+import android.media.MediaRecorder
+import android.util.Log
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.io.File
+import java.io.IOException
+
+class AudioRecorder(
+    private val applicationContext: Context
+) : IAudioRecorder {
+    // default recording params
+    private var recordingParams: RecordingParams = RecordingParams(
+        maxDuration = 60 * 5, // 5 minutes
+        maxFileSize = 1000000L * 25 // 25MB
+    )
+
+    private var onRecordingFinished: (String) -> Unit = {}
+
+    // todo: check place of the recording file
+    private val storeInMemory = true
+    private val filePath by lazy {
+        if (storeInMemory) {
+            applicationContext.cacheDir.absolutePath + "/recording.3gp"
```

Comment:
@pantstamp I didn't think this media type is supported by the endpoint? PCYsg-WLi-p2 . I believe it's working, but not listed as an official type. Not sure if this will come back to bite us. I'll check the BE code.

Reply:
I changed it to mp4 that is officially supported. Tested it and it seem that it works.

Reply:
Double checked the BE too.
```
$allowed_mime_types = array(
			'audio/mpeg', // mp3, mpeg, mpga files
			'audio/mp3', // alternative mp3 files
			'audio/ogg', // ogg files
			'audio/flac', // flac files
			'audio/x-flac', // alternative flac files
			'audio/m4a', // m4a files
			'audio/x-m4a', // alternative m4a files
			'audio/mp4', // alternative m4a files
			'audio/wav', // wav files
			'audio/wave', // alternative wav files
			'audio/x-wav', // alternative wav files
			'audio/webm', // webm files
		);
```